vision/mlp_mnist/README.md

```diff
@@ -6,10 +6,20 @@
 
 ## Model Info
 
-An MLP consists of at least three of nodes: an input layer, a hidden layer and an output layer. Except for the input node each node is a neuron that uses a nonlinear activation function. MLP utilizes a supervised learning technique called backpropagation for training. Its multiple layers and non-linear activation distinguish MLP from a linear perceptron. It can distinguish data that is not linearly separable.
+A [multi-layer perceptron](https://en.wikipedia.org/wiki/Multilayer_perceptron) (MLP) consists of at least three sets of of nodes: an input layer, a hidden layer and an output layer. Except for the input node each node is a neuron that uses a nonlinear activation function. MLP utilizes a supervised learning technique called backpropagation for training. Its multiple layers and non-linear activation distinguish MLP from a linear perceptron. It can distinguish data that is not linearly separable.
+
+In this example, we create a MLP that classifies handwritten digits using the [MNIST dataset](http://yann.lecun.com/exdb/mnist/).
+Our model uses the simplest [Flux layers](http://fluxml.ai/Flux.jl/stable/models/layers/), namely `Dense` and `Chain`.
+Since it uses [softmax](https://en.wikipedia.org/wiki/Softmax_function) on its outputs, and [`crossentopy`](http://fluxml.ai/Flux.jl/stable/models/losses/#Flux.Losses.crossentropy) as the loss function.
+
+For simplicity this model does not use a graphics card, since an ordinary CPU is fast enough.
+See for example the [LeNet convolutional network](https://github.com/FluxML/model-zoo/tree/master/vision/conv_mnist) for GPU usage.
 
 ## Training
 
+You can copy and paste the example into the Julia REPL to see what each part does.
+Or you can run it all at once from the terminal, like this:
```

Comment:
Running all at once from the terminal won't show any output. Could we use `print` on the predictions and display the image even if users aren't running from the REPL?

Reply:
For me it does print:
```
mlp_mnist % /Applications/Julia-1.9.app/Contents/Resources/julia/bin/julia --project mlp_mnist.jl
┌ Info: After epoch = 1
│   loss = 1.8101638024672866
│   train_acc = 75.81
└   test_acc = 76.75
┌ Info: After epoch = 3
│   loss = 0.8674804433248937
│   train_acc = 85.65
└   test_acc = 86.22
```

Reply:
If not the images, perhaps we could print out the predictions/ground truth labels in lines 110+115 for comparison.